src/mongocxx/test/spec/unified_tests/runner.cpp

```diff
@@ -220,8 +220,7 @@ bool equals_server_topology(const document::element& topologies) {
     // test file.
     const static std::string actual = test_util::get_topology();
     const auto equals = [&](const bsoncxx::array::element& expected) {
-        return expected == value(actual) ||
-               (expected == value("sharded") && actual == "sharded-replicaset");
+        return expected == value(actual);
```

Comment:
I suggest reverting the changes in the .cpp files for handling `sharded-replicaset`. The `sharded-replicaset` value is [deprecated but not removed](https://github.com/mongodb/specifications/blob/840e6d49c354656bff11b2622f0d3001b39d9403/source/unified-test-format/unified-test-format.rst) from the Unified Test Format:

> The "sharded-replicaset" topology type is deprecated. MongoDB 3.6+ requires that all shard servers be replica sets (see: [release notes](https://www.mongodb.com/docs/manual/release-notes/3.6-upgrade-sharded-cluster/#shard-replica-sets)). Therefore, tests SHOULD use "sharded" instead of "sharded-replicaset" when targeting 3.6+ server versions in order to avoid unnecessary overhead.

The [runCommand.yml test](https://github.com/mongodb/specifications/blob/840e6d49c354656bff11b2622f0d3001b39d9403/source/run-command/tests/unified/runCommand.yml#L122) still references sharded-replicaset.